app/services/checkout/form_data_adapter.rb

```diff
@@ -6,7 +6,7 @@ class FormDataAdapter
     attr_reader :params, :shipping_method_id
 
     def initialize(params, order, current_user)
-      @params = params.dup
+      @params = params.deep_dup.to_h.with_indifferent_access
```

Comment:
This feels a little ... clunky? Brittle? Not great :) But I don't have a better suggestion, except maybe that whatever is calling this method should be expected to pass in a hash with indifferent access.

Reply:
This class does a hell of a lot of mangling of the params object and expects it to work as it did in Rails 4. I think part of the issue here is that we're passing a real params object in some tests and a mock params object in other tests, and we need the object to be more-or-less uniform in any context (and it needs to behave like a Rails 4 params object). 

Yeah, it's not pretty. Actually now that I look again, I think the `#deep_dup` might be unnecessary here...